tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -954,25 +932,23 @@ public void onContainersCompleted(List<ContainerStatus> completedContainers) {
           LOG.info(diagnostics);
         }
         TonyTask task = session.getTask(containerStatus.getContainerId());
+
         if (task != null) {
+          // Ignore tasks from past sessions.
+          if (task.getSessionId() != session.sessionId) {
+            return;
+          }
+          // Update TensorFlowSession on the state of the task.
+          session.onTaskCompleted(task.getJobName(), task.getTaskIndex(), containerStatus.getExitStatus());
```

Comment:
nit: containerStatus.getExitStatus() -> exitStatus
there's another access on line 929, if you could change that as well that'd be great :)